File: java/ex6.java
public class ex6 {
    static class Motor {
        private String tipo; 
        private int potencia; 
    
        public Motor(String tipo, int potencia) {
            this.tipo = tipo;
            this.potencia = potencia;
        }
    
        public String getTipo() {
            return tipo;
        }
    
        public int getPotencia() {
            return potencia;
        }
    
        public void mostrarDetalhes() {
            System.out.println("Motor tipo: " + tipo + ", Potência: " + potencia + " CV");
        }
    }
    
    static class Veiculo {
        private String marca;
        private String modelo;
        private int ano;
        private Motor motor; 
    
        public Veiculo(String marca, String modelo, int ano, Motor motor) {
            this.marca = marca;
            this.modelo = modelo;
            this.ano = ano;
            this.motor = motor; 
        }
    
        public void mostrarDetalhes() {
            System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
            motor.mostrarDetalhes(); 
        }
    
        public static void main(String[] args) {
            
            Motor motor1 = new Motor("Gasolina", 120);
    
            Veiculo veiculo1 = new Veiculo("Toyota", "Corolla", 2020, motor1);
    
            veiculo1.mostrarDetalhes();
        }
    }
    
}

File: golang/ex6.go
package main

import (
	"fmt"
)

type Propulsor struct {
	Potencia int    // Potência do motor em cavalos de potência (CV)
	Tipo     string // Tipo de motor (ex: "Gasolina", "Álcool", "Elétrico")
}

type Veiculo struct {
	Marca  string
	Modelo string
	Ano    int
	Propulsor Propulsor // Associando um objeto Propulsor à classe Veiculo
}

func main() {
	propulsor1 := Propulsor{Potencia: 150, Tipo: "Gasolina"}
	veiculo1 := Veiculo{Marca: "Toyota", Modelo: "Corolla", Ano: 2020, Propulsor: propulsor1}

	propulsor2 := Propulsor{Potencia: 200, Tipo: "Elétrico"}
	veiculo2 := Veiculo{Marca: "Tesla", Modelo: "Model 3", Ano: 2021, Propulsor: propulsor2}

	fmt.Printf("Veículo: %s %s (%d) - Propulsor: %d CV, Tipo: %s\n", veiculo1.Marca, veiculo1.Modelo, veiculo1.Ano, veiculo1.Propulsor.Potencia, veiculo1.Propulsor.Tipo)
	fmt.Printf("Veículo: %s %s (%d) - Propulsor: %d CV, Tipo: %s\n", veiculo2.Marca, veiculo2.Modelo, veiculo2.Ano, veiculo2.Propulsor.Potencia, veiculo2.Propulsor.Tipo)
}
